backend/src/main/java/com/ambacam/service/RequeteService.java

```diff
@@ -291,6 +297,53 @@ public Requete updateStatus(Long id, RequeteStatusTO requeteStatusTO) {
 		return requeteRepository.save(found);
 	}
 
+	/**
+	 * Update status of some Requetes in a RequeteGroupe
+	 * 
+	 * @param requeteGroupeId
+	 * @param assignStatusTO
+	 * 
+	 * @throws ResourceNotFoundException
+         *             if the requeteGroupe is not found
+         * @throws ResourceBadRequestException
+         *             if the given statusId does not exist
+         * @throws ResourceBadRequestException
+         *             if the requete groupe does not contains a requete
+         * @throws ResourceBadRequestException
+         *             if a requete to update does not exist       
+         * @throws ResourceBadRequestException
+         *             if a requete already has the given status    
+	 */
+	public void assignStatus(Long requeteGroupeId, AssignStatusTO assignStatusTO) {
+	    // find requete groupe
+	    RequeteGroupe requeteGroupe = findRequeteGroupe(requeteGroupeId);
+	    
+	    // find status
+	    StatusRequete status = findStatusRequete(assignStatusTO.getStatusId());
+	    
+	    // find requetes
+	    List<Requete> requetes = requeteRepository.findAllByRequeteGroupe(requeteGroupe);
+	    
+	    assignStatusTO.getRequeteIds().forEach(requeteId -> {
```

Comment:
Maybe the swagger was understood wrong. The transfer object here is a RequeteStatusTO, which is just about transporting the status id, all the requests belonging to the group should be assigned with.
From that point of view, I believe that the implementation here should just update the requests belonging to this requeteGroupe with the specified status.
Please take a look at the swagger again.

Is there any reason why you implemented it that way?

I believe that the function responsible for that should be in the `RequeteGroupeService` and call the function `RequeteService.updateStatus()`.

Reply:
This function is the implementation of the service for the endpoint: 
"/operateurs/{operateurId}/requete-groupes/{requeteGroupeId}/requetes/status"
Please take a look at the swagger regarding this endpoint. Maybe the specification is wrong.

Reply:
You are right! Thanks for that.